crawler/src/main.rs

```diff
@@ -0,0 +1,72 @@
+use colored::*;
+use serde::{Deserialize, Serialize};
+
+type Listing = ProxyData<Posts>;
+
+#[derive(Debug, Deserialize, Serialize)]
+struct ProxyData<T> {
+    pub kind: String,
+    pub data: T,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct Posts {
+    pub children: Vec<ProxyData<Post>>,
+    pub after: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct Post {
+    pub is_self: bool,
+    pub url: String,
+    pub subreddit: String,
```

Comment:
Why parse `subreddit` (and `ProxyData::kind`) if it isn't used?

Reply:
No reason